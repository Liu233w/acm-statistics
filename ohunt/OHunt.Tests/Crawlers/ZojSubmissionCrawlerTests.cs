using System.Threading;
using System.Threading.Tasks;
using System.Threading.Tasks.Dataflow;
using FluentAssertions;
using Microsoft.Extensions.Logging;
using Moq;
using OHunt.Web.Crawlers;
using OHunt.Web.Models;
using Snapshooter.Xunit;
using Xunit;

namespace OHunt.Tests.Crawlers
{
    [Trait("Category", "WithNetwork")]
    public class ZojSubmissionCrawlerTests
    {
        private readonly ILogger<ZojSubmissionCrawler> _loggerMock =
            Mock.Of<ILogger<ZojSubmissionCrawler>>();

        [Fact]
        public async Task It_ShouldGetCorrectResult()
        {
            var crawler = new ZojSubmissionCrawler(_loggerMock);

            var messages
                = new BufferBlock<CrawlerMessage>(new DataflowBlockOptions
                {
                    EnsureOrdered = true,
                });

#pragma warning disable 4014
            crawler.WorkAsync(null, messages, new CancellationToken());
#pragma warning restore 4014

            const int bufferLength = 110;

            var list = new Submission[bufferLength];
            for (var i = 0; i < bufferLength; i++)
            {
                var message = await messages.ReceiveAsync();
                message.CrawlerError.Should().BeNull();
                message.IsCheckPoint.Should().BeTrue();
                list[i] = message.Submission;
            }

            Snapshot.Match(list);
        }
    }
}
