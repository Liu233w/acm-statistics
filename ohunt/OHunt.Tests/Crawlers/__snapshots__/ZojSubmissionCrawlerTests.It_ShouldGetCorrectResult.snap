[
  {
    "SubmissionId": 91827364599,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "Accepted",
    "ProblemLabel": "1049",
    "Time": "2005-09-13T13:53:37+10:00"
  },
  {
    "SubmissionId": 91827364598,
    "OnlineJudgeId": "ZOJ",
    "UserName": "timgreen",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:53:15+10:00"
  },
  {
    "SubmissionId": 91827364597,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1049",
    "Time": "2005-09-13T13:53:08+10:00"
  },
  {
    "SubmissionId": 91827364596,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "Accepted",
    "ProblemLabel": "1048",
    "Time": "2005-09-13T13:53:06+10:00"
  },
  {
    "SubmissionId": 91827364595,
    "OnlineJudgeId": "ZOJ",
    "UserName": "timgreen",
    "Status": "CompileError",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:52:47+10:00"
  },
  {
    "SubmissionId": 91827364594,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1048",
    "Time": "2005-09-13T13:52:45+10:00"
  },
  {
    "SubmissionId": 91827364593,
    "OnlineJudgeId": "ZOJ",
    "UserName": "archenxu",
    "Status": "UnknownError",
    "ProblemLabel": "1538",
    "Time": "2005-09-13T13:52:44+10:00"
  },
  {
    "SubmissionId": 91827364592,
    "OnlineJudgeId": "ZOJ",
    "UserName": "saltycookie",
    "Status": "Accepted",
    "ProblemLabel": "2215",
    "Time": "2005-09-13T13:52:40+10:00"
  },
  {
    "SubmissionId": 91827364591,
    "OnlineJudgeId": "ZOJ",
    "UserName": "yanshan",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:52:38+10:00"
  },
  {
    "SubmissionId": 91827364590,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "WrongAnswer",
    "ProblemLabel": "1006",
    "Time": "2005-09-13T13:52:32+10:00"
  },
  {
    "SubmissionId": 91827364589,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:51:45+10:00"
  },
  {
    "SubmissionId": 91827364588,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1047",
    "Time": "2005-09-13T13:51:33+10:00"
  },
  {
    "SubmissionId": 91827364587,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1045",
    "Time": "2005-09-13T13:51:09+10:00"
  },
  {
    "SubmissionId": 91827364586,
    "OnlineJudgeId": "ZOJ",
    "UserName": "saltycookie",
    "Status": "UnknownError",
    "ProblemLabel": "2213",
    "Time": "2005-09-13T13:50:49+10:00"
  },
  {
    "SubmissionId": 91827364585,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1042",
    "Time": "2005-09-13T13:50:36+10:00"
  },
  {
    "SubmissionId": 91827364584,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1041",
    "Time": "2005-09-13T13:50:16+10:00"
  },
  {
    "SubmissionId": 91827364583,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1037",
    "Time": "2005-09-13T13:49:49+10:00"
  },
  {
    "SubmissionId": 91827364582,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "WrongAnswer",
    "ProblemLabel": "1006",
    "Time": "2005-09-13T13:49:37+10:00"
  },
  {
    "SubmissionId": 91827364581,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1029",
    "Time": "2005-09-13T13:49:19+10:00"
  },
  {
    "SubmissionId": 91827364580,
    "OnlineJudgeId": "ZOJ",
    "UserName": "saltycookie",
    "Status": "Accepted",
    "ProblemLabel": "2307",
    "Time": "2005-09-13T13:49:13+10:00"
  },
  {
    "SubmissionId": 91827364579,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1027",
    "Time": "2005-09-13T13:48:57+10:00"
  },
  {
    "SubmissionId": 91827364578,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1026",
    "Time": "2005-09-13T13:48:29+10:00"
  },
  {
    "SubmissionId": 91827364577,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "2133",
    "Time": "2005-09-13T13:48:24+10:00"
  },
  {
    "SubmissionId": 91827364576,
    "OnlineJudgeId": "ZOJ",
    "UserName": "saltycookie",
    "Status": "Accepted",
    "ProblemLabel": "2308",
    "Time": "2005-09-13T13:48:11+10:00"
  },
  {
    "SubmissionId": 91827364575,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "WrongAnswer",
    "ProblemLabel": "1023",
    "Time": "2005-09-13T13:47:46+10:00"
  },
  {
    "SubmissionId": 91827364574,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "CompileError",
    "ProblemLabel": "1562",
    "Time": "2005-09-13T13:47:10+10:00"
  },
  {
    "SubmissionId": 91827364573,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1020",
    "Time": "2005-09-13T13:46:39+10:00"
  },
  {
    "SubmissionId": 91827364572,
    "OnlineJudgeId": "ZOJ",
    "UserName": "sothink",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:46:36+10:00"
  },
  {
    "SubmissionId": 91827364571,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1008",
    "Time": "2005-09-13T13:46:07+10:00"
  },
  {
    "SubmissionId": 91827364570,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1010",
    "Time": "2005-09-13T13:45:44+10:00"
  },
  {
    "SubmissionId": 91827364569,
    "OnlineJudgeId": "ZOJ",
    "UserName": "saltycookie",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:45:41+10:00"
  },
  {
    "SubmissionId": 91827364568,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1011",
    "Time": "2005-09-13T13:45:25+10:00"
  },
  {
    "SubmissionId": 91827364567,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "UnknownError",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:44:45+10:00"
  },
  {
    "SubmissionId": 91827364566,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "CompileError",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:43:58+10:00"
  },
  {
    "SubmissionId": 91827364565,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "CompileError",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:43:25+10:00"
  },
  {
    "SubmissionId": 91827364564,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "WrongAnswer",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:43:10+10:00"
  },
  {
    "SubmissionId": 91827364563,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "CompileError",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:42:34+10:00"
  },
  {
    "SubmissionId": 91827364562,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "CompileError",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:41:52+10:00"
  },
  {
    "SubmissionId": 91827364561,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "CompileError",
    "ProblemLabel": "1014",
    "Time": "2005-09-13T13:41:06+10:00"
  },
  {
    "SubmissionId": 91827364560,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1013",
    "Time": "2005-09-13T13:40:40+10:00"
  },
  {
    "SubmissionId": 91827364559,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1003",
    "Time": "2005-09-13T13:40:27+10:00"
  },
  {
    "SubmissionId": 91827364558,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "UnknownError",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:38:56+10:00"
  },
  {
    "SubmissionId": 91827364557,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "UnknownError",
    "ProblemLabel": "1015",
    "Time": "2005-09-13T13:38:46+10:00"
  },
  {
    "SubmissionId": 91827364556,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "UnknownError",
    "ProblemLabel": "1010",
    "Time": "2005-09-13T13:38:20+10:00"
  },
  {
    "SubmissionId": 91827364555,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "UnknownError",
    "ProblemLabel": "1010",
    "Time": "2005-09-13T13:38:11+10:00"
  },
  {
    "SubmissionId": 91827364554,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "UnknownError",
    "ProblemLabel": "1010",
    "Time": "2005-09-13T13:37:59+10:00"
  },
  {
    "SubmissionId": 91827364553,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1009",
    "Time": "2005-09-13T13:37:35+10:00"
  },
  {
    "SubmissionId": 91827364552,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1006",
    "Time": "2005-09-13T13:37:18+10:00"
  },
  {
    "SubmissionId": 91827364551,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1002",
    "Time": "2005-09-13T13:37:02+10:00"
  },
  {
    "SubmissionId": 91827364550,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:36:42+10:00"
  },
  {
    "SubmissionId": 91827364549,
    "OnlineJudgeId": "ZOJ",
    "UserName": "DXDSpirits",
    "Status": "CompileError",
    "ProblemLabel": "1562",
    "Time": "2005-09-13T13:35:49+10:00"
  },
  {
    "SubmissionId": 91827364548,
    "OnlineJudgeId": "ZOJ",
    "UserName": "leftleaf",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:33:20+10:00"
  },
  {
    "SubmissionId": 91827364547,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1007",
    "Time": "2005-09-13T13:29:53+10:00"
  },
  {
    "SubmissionId": 91827364546,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1005",
    "Time": "2005-09-13T13:29:14+10:00"
  },
  {
    "SubmissionId": 91827364545,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1004",
    "Time": "2005-09-13T13:28:51+10:00"
  },
  {
    "SubmissionId": 91827364544,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1003",
    "Time": "2005-09-13T13:28:31+10:00"
  },
  {
    "SubmissionId": 91827364543,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1002",
    "Time": "2005-09-13T13:27:38+10:00"
  },
  {
    "SubmissionId": 91827364542,
    "OnlineJudgeId": "ZOJ",
    "UserName": "jyy",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:26:11+10:00"
  },
  {
    "SubmissionId": 91827364541,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:25:09+10:00"
  },
  {
    "SubmissionId": 91827364540,
    "OnlineJudgeId": "ZOJ",
    "UserName": "yanshan",
    "Status": "WrongAnswer",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:23:34+10:00"
  },
  {
    "SubmissionId": 91827364539,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "2263",
    "Time": "2005-09-13T13:22:31+10:00"
  },
  {
    "SubmissionId": 91827364538,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "WrongAnswer",
    "ProblemLabel": "2293",
    "Time": "2005-09-13T13:20:53+10:00"
  },
  {
    "SubmissionId": 91827364537,
    "OnlineJudgeId": "ZOJ",
    "UserName": "skatou",
    "Status": "Accepted",
    "ProblemLabel": "2399",
    "Time": "2005-09-13T13:20:01+10:00"
  },
  {
    "SubmissionId": 91827364536,
    "OnlineJudgeId": "ZOJ",
    "UserName": "skatou",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T13:19:35+10:00"
  },
  {
    "SubmissionId": 91827364535,
    "OnlineJudgeId": "ZOJ",
    "UserName": "suninth",
    "Status": "UnknownError",
    "ProblemLabel": "1907",
    "Time": "2005-09-13T13:19:22+10:00"
  },
  {
    "SubmissionId": 91827364534,
    "OnlineJudgeId": "ZOJ",
    "UserName": "gougou",
    "Status": "Accepted",
    "ProblemLabel": "1004",
    "Time": "2005-09-13T13:18:55+10:00"
  },
  {
    "SubmissionId": 91827364533,
    "OnlineJudgeId": "ZOJ",
    "UserName": "suninth",
    "Status": "CompileError",
    "ProblemLabel": "1907",
    "Time": "2005-09-13T13:18:32+10:00"
  },
  {
    "SubmissionId": 91827364532,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lala",
    "Status": "Accepted",
    "ProblemLabel": "1004",
    "Time": "2005-09-13T13:13:59+10:00"
  },
  {
    "SubmissionId": 91827364531,
    "OnlineJudgeId": "ZOJ",
    "UserName": "riveria",
    "Status": "Accepted",
    "ProblemLabel": "1004",
    "Time": "2005-09-13T13:11:58+10:00"
  },
  {
    "SubmissionId": 91827364530,
    "OnlineJudgeId": "ZOJ",
    "UserName": "riveria",
    "Status": "CompileError",
    "ProblemLabel": "1004",
    "Time": "2005-09-13T13:11:31+10:00"
  },
  {
    "SubmissionId": 91827364529,
    "OnlineJudgeId": "ZOJ",
    "UserName": "riveria",
    "Status": "Accepted",
    "ProblemLabel": "1003",
    "Time": "2005-09-13T13:10:56+10:00"
  },
  {
    "SubmissionId": 91827364528,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "WrongAnswer",
    "ProblemLabel": "1026",
    "Time": "2005-09-13T13:00:38+10:00"
  },
  {
    "SubmissionId": 91827364527,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1025",
    "Time": "2005-09-13T13:00:16+10:00"
  },
  {
    "SubmissionId": 91827364526,
    "OnlineJudgeId": "ZOJ",
    "UserName": "MyheartFly",
    "Status": "WrongAnswer",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:59:49+10:00"
  },
  {
    "SubmissionId": 91827364525,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1016",
    "Time": "2005-09-13T12:59:47+10:00"
  },
  {
    "SubmissionId": 91827364524,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1009",
    "Time": "2005-09-13T12:59:16+10:00"
  },
  {
    "SubmissionId": 91827364523,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "UnknownError",
    "ProblemLabel": "1008",
    "Time": "2005-09-13T12:58:44+10:00"
  },
  {
    "SubmissionId": 91827364522,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1007",
    "Time": "2005-09-13T12:58:20+10:00"
  },
  {
    "SubmissionId": 91827364521,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1006",
    "Time": "2005-09-13T12:58:02+10:00"
  },
  {
    "SubmissionId": 91827364520,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1005",
    "Time": "2005-09-13T12:57:39+10:00"
  },
  {
    "SubmissionId": 91827364519,
    "OnlineJudgeId": "ZOJ",
    "UserName": "ylzhang",
    "Status": "CompileError",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:57:34+10:00"
  },
  {
    "SubmissionId": 91827364518,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1004",
    "Time": "2005-09-13T12:57:17+10:00"
  },
  {
    "SubmissionId": 91827364517,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1003",
    "Time": "2005-09-13T12:56:48+10:00"
  },
  {
    "SubmissionId": 91827364516,
    "OnlineJudgeId": "ZOJ",
    "UserName": "bluethinking",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:56:47+10:00"
  },
  {
    "SubmissionId": 91827364515,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1002",
    "Time": "2005-09-13T12:56:19+10:00"
  },
  {
    "SubmissionId": 91827364514,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caojia321",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:55:51+10:00"
  },
  {
    "SubmissionId": 91827364513,
    "OnlineJudgeId": "ZOJ",
    "UserName": "bluethinking",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:53:34+10:00"
  },
  {
    "SubmissionId": 91827364512,
    "OnlineJudgeId": "ZOJ",
    "UserName": "feng5166",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:46:45+10:00"
  },
  {
    "SubmissionId": 91827364511,
    "OnlineJudgeId": "ZOJ",
    "UserName": "wangzhipeng",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:46:24+10:00"
  },
  {
    "SubmissionId": 91827364510,
    "OnlineJudgeId": "ZOJ",
    "UserName": "riveria",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:46:06+10:00"
  },
  {
    "SubmissionId": 91827364509,
    "OnlineJudgeId": "ZOJ",
    "UserName": "riveria",
    "Status": "UnknownError",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T12:44:38+10:00"
  },
  {
    "SubmissionId": 91827364508,
    "OnlineJudgeId": "ZOJ",
    "UserName": "riveria",
    "Status": "UnknownError",
    "ProblemLabel": "1001",
    "Time": "2005-09-12T12:53:00+10:00"
  },
  {
    "SubmissionId": 91827364507,
    "OnlineJudgeId": "ZOJ",
    "UserName": "hawk",
    "Status": "Accepted",
    "ProblemLabel": "1048",
    "Time": "2005-09-17T13:38:56+10:00"
  },
  {
    "SubmissionId": 91827364506,
    "OnlineJudgeId": "ZOJ",
    "UserName": "rock",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-17T13:39:26+10:00"
  },
  {
    "SubmissionId": 91827364505,
    "OnlineJudgeId": "ZOJ",
    "UserName": "judas",
    "Status": "WrongAnswer",
    "ProblemLabel": "1489",
    "Time": "2005-09-17T13:39:31+10:00"
  },
  {
    "SubmissionId": 91827364504,
    "OnlineJudgeId": "ZOJ",
    "UserName": "rock",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-17T13:39:37+10:00"
  },
  {
    "SubmissionId": 91827364503,
    "OnlineJudgeId": "ZOJ",
    "UserName": "xemo",
    "Status": "Accepted",
    "ProblemLabel": "1905",
    "Time": "2005-09-17T13:41:20+10:00"
  },
  {
    "SubmissionId": 91827364502,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lovekobe",
    "Status": "CompileError",
    "ProblemLabel": "1115",
    "Time": "2005-09-17T13:42:17+10:00"
  },
  {
    "SubmissionId": 91827364501,
    "OnlineJudgeId": "ZOJ",
    "UserName": "xemo",
    "Status": "Accepted",
    "ProblemLabel": "1905",
    "Time": "2005-09-17T13:43:11+10:00"
  },
  {
    "SubmissionId": 91827364500,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lovekobe",
    "Status": "CompileError",
    "ProblemLabel": "1115",
    "Time": "2005-09-17T13:45:40+10:00"
  },
  {
    "SubmissionId": 91827364699,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "Accepted",
    "ProblemLabel": "1365",
    "Time": "2005-09-13T14:14:40+10:00"
  },
  {
    "SubmissionId": 91827364698,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "Accepted",
    "ProblemLabel": "2388",
    "Time": "2005-09-13T14:14:01+10:00"
  },
  {
    "SubmissionId": 91827364697,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "WrongAnswer",
    "ProblemLabel": "2060",
    "Time": "2005-09-13T14:13:24+10:00"
  },
  {
    "SubmissionId": 91827364696,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1037",
    "Time": "2005-09-13T14:13:21+10:00"
  },
  {
    "SubmissionId": 91827364695,
    "OnlineJudgeId": "ZOJ",
    "UserName": "tinystar",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T14:13:06+10:00"
  },
  {
    "SubmissionId": 91827364694,
    "OnlineJudgeId": "ZOJ",
    "UserName": "andy",
    "Status": "Accepted",
    "ProblemLabel": "1196",
    "Time": "2005-09-13T14:12:42+10:00"
  },
  {
    "SubmissionId": 91827364693,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1038",
    "Time": "2005-09-13T14:12:32+10:00"
  },
  {
    "SubmissionId": 91827364692,
    "OnlineJudgeId": "ZOJ",
    "UserName": "lily",
    "Status": "Accepted",
    "ProblemLabel": "2104",
    "Time": "2005-09-13T14:12:32+10:00"
  },
  {
    "SubmissionId": 91827364691,
    "OnlineJudgeId": "ZOJ",
    "UserName": "xingning001",
    "Status": "Accepted",
    "ProblemLabel": "1001",
    "Time": "2005-09-13T14:12:17+10:00"
  },
  {
    "SubmissionId": 91827364690,
    "OnlineJudgeId": "ZOJ",
    "UserName": "caopeng",
    "Status": "Accepted",
    "ProblemLabel": "1034",
    "Time": "2005-09-13T14:12:11+10:00"
  }
]
